scripts/pdf/generate_pdf.py

```diff
@@ -1,225 +1,63 @@
-            )
-        elif ".mdx" in entry or ".md" in entry:
-            allFiles.append(ToCItem(fullPath, parentChapter + str(chapter)))
-
-        chapter += 1
-    return allFiles
-
-
-def main():
-    dirName = ""
-    try:
-        dirName = sys.argv[1]
-        dirName = re.sub(r"\/$", "", dirName)
-    except BaseException:
-        print("directory not passed in")
-        print("if running from yarn use `yarn build-pdf directory/path/here`")
-        sys.exit(1)
-
-    openPdf = False
-    html = False
-    try:
-        html = sys.argv[2] == "--html"
-        openPdf = sys.argv[2] == "--open"
-    except BaseException:
-        pass
-
-    splitDirName = dirName.split("/")
-    product = splitDirName[2]
-    version = splitDirName[3]
-
-    fullProductPdf = True
-    guide = None
-    if len(splitDirName) > 4:
-        fullProductPdf = False
-        guide = splitDirName[4]
-
-    _file_prefix = f"{dirName}/{product}_v{version}"
-    _doc_prefix = f"{_file_prefix}_documentation"
-    mdxFilePath = f"{_doc_prefix}.mdx"
-    htmlFilePath = f"{_doc_prefix}.html"
-    coverFilePath = f"{_doc_prefix}_cover.html"
-    pdfFilePath = f"{_file_prefix}" "_{}documentation.pdf".format(
-        guide + "_" if guide else ""
-    )
+    print(f"{ANSI_BLUE}building {pdf_file}{ANSI_STOP}")
+    with open(mdx_file, "w") as output:
+        resource_search_paths = {doc_path, *combine_mdx(files, output)}
 
-    print(f"{ANSI_BLUE}building {pdfFilePath}{ANSI_STOP}")
-
-    if not os.path.exists(dirName):
-        raise Exception("directory does not exist")
-
-    if os.path.exists(mdxFilePath):
-        os.remove(mdxFilePath)
-
-    # Get the list of all files in directory tree at given path
-    listOfFiles = getListOfFiles(dirName, "")
-    if len(listOfFiles) == 0:
-        raise Exception(f"no files in {dirName}")
-    if fullProductPdf:
-        listOfFiles.pop(0)  # remove base product index page, which are empty
-
-    for elem in listOfFiles:
-        g = open(elem.filename, "r")
-        for line in g.readlines():
-            if "title: " in line:
-                elem.title = line[7:].replace('"', "").replace("\n", "")
-            pattern = re.compile('div id="(.*?)" class="registered_link"')
-
-            tag = pattern.search(line)
-            if tag and len(elem.anchor) == 0:
-                elem.anchor = tag.group(1)
-
-    resourceSearchPaths = {dirName}
-
-    # Print the files
-    with open(mdxFilePath, "w") as fp:
-        for elem in listOfFiles:
-            g = open(elem.filename, "r")
-
-            baseImagePath = os.path.split(elem.filename)[0]
-            resourceSearchPaths.add(baseImagePath)
-
-            frontmatterCount = 2
-            for line in g.readlines():
-                newLine = line
-
-                if line[0:2] == "# ":
-                    newLine = "##" + line
-                if line[0:3] == "## ":
-                    newLine = "#" + line
```

Comment:
@josh-heyer @robert-stringer Is there a reason why we only increment the chapter levels for h1 and h2 in PDFs?

I would think we'd want to increase the chapter levels universally, regardless of levels.

Reply:
Only reason I can think of is to reduce the length & depth of the ToC; could potentially get pretty big if there are a lot of pages with numerous h3 (etc) sections.

Reply:
The headings do not contribute to TOC. Those are at the file level and gotten from the frontmatter. I'm going to add an additional indent level for everything, not just h1 and h2. This way the PDF will have some consistency with the web content.